Content.Server/Blob/BlobFactorySystem.cs

```diff
@@ -0,0 +1,96 @@
+using Content.Server.Blob.NPC.BlobPod;
+using Content.Server.Fluids.EntitySystems;
+using Content.Shared.Blob;
+using Content.Shared.Damage;
+using Content.Shared.Destructible;
+using Content.Shared.Weapons.Melee;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Blob;
+
+public sealed class BlobFactorySystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<BlobFactoryComponent, ComponentStartup>(OnStartup);
+        SubscribeLocalEvent<BlobFactoryComponent, BlobTileGetPulseEvent>(OnPulsed);
+        SubscribeLocalEvent<BlobFactoryComponent, ProduceBlobbernautEvent>(OnProduceBlobbernaut);
+        SubscribeLocalEvent<BlobFactoryComponent, DestructionEventArgs>(OnDestruction);
+    }
+
+    private void OnStartup(EntityUid uid, BlobFactoryComponent observerComponent, ComponentStartup args)
+    {
+
+    }
+
+    private void OnDestruction(EntityUid uid, BlobFactoryComponent component, DestructionEventArgs args)
+    {
+        if (TryComp<BlobbernautComponent>(component.Blobbernaut, out var blobbernautComponent))
+        {
+            blobbernautComponent.Factory = null;
+        }
+    }
+
+    private void OnProduceBlobbernaut(EntityUid uid, BlobFactoryComponent component, ProduceBlobbernautEvent args)
+    {
+        if (component.Blobbernaut != null)
+            return;
+
+        if (!TryComp<BlobTileComponent>(uid, out var blobTileComponent) || blobTileComponent.Core == null)
+            return;
+
+        if (!TryComp<BlobCoreComponent>(blobTileComponent.Core, out var blobCoreComponent))
+            return;
+
+        var xform = Transform(uid);
+
+        var blobbernaut = Spawn(component.BlobbernautId, xform.Coordinates);
+
+        component.Blobbernaut = blobbernaut;
+        if (TryComp<BlobbernautComponent>(blobbernaut, out var blobbernautComponent))
+        {
+            blobbernautComponent.Factory = uid;
+            blobbernautComponent.Color = blobCoreComponent.ChemСolors[blobCoreComponent.CurrentChem];
+            Dirty(blobbernautComponent);
+        }
+        if (TryComp<MeleeWeaponComponent>(blobbernaut, out var meleeWeaponComponent))
```

Comment:
Blobbernaut should handle damage on it's own, the factory doesn't have anything to do with it.

Create a SpawnBlobbernaut helper for creating the blobbernaut, and then let the blobbernaut configure it's own damage effects in *its* system.